runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -298,30 +298,43 @@ public UUID getStreamID() {
     }
 
     private <R> R TXExecuteInner(Supplier<R> txFunction, boolean isMetricsEnabled) {
-        long sleepTime = 1L;
         final long maxSleepTime = 1000L;
+        long sleepTime = 1L;
         int retries = 1;
+        int maxNumOfRetries = 20; // Approximately 10 seconds.
```

Comment:
I am not sure how to do it cleanly. It's either we:

1) Pass the value all the way to CompileProxy from the Runtime that requires changing constructor signatures for a couple of classes. (ugly)
2) Make the parameter static (somewhat less ugly but Maithem doesn't like this approach).